Add API to respond with for running backfills

jrobotham-square opened the pull request:
When an incident occurs it's often really valuable to be able to see what Backfills are currently running in order to:
- Identify ones that may be causing or exacerbating issues.
- Identify ones that could be stopped to reduce the risk.

This PR exposes a new endpoint to surface the currently running backfills.

As part of this change I've also included instructions on starting mysql locally for development and fixed some issues that occur locally when building / running tests on Apple silicon.

Subsequent PR will integrate this endpoint into the UI to create a screen something like:
![Screenshot 2023-11-09 at 8 37 16 am](https://github.com/cashapp/backfila/assets/92703335/827ac575-7927-4a34-8495-b303971bcfd2)

adrw commented:
Mind reposting the screenshots, they don't seem to be showing in the preview?

mpawliszyn commented:
Make sure to rebase on top of the variant PR.

mpawliszyn commented on service/src/main/kotlin/app/cash/backfila/dashboard/GetActiveBackfillsAction.kt:
```diff
@@ -0,0 +1,64 @@
+  private val queryFactory: Query.Factory,
+) : WebAction {
+
+  @Get("/backfills/running")
```
Probably needs to be paginated.

jrobotham-square replied:
yeah - was kinda assuming the number of backfills concurrently running wouldn't generally be enough to require pagination, but that's I guess that's not necessarily the case.

jrobotham-square replied:
Actually, looks like we don't have pagination on any of the endpoints apart from the `/services/{service}/variants/{variant}/backfill-runs` one.  (which makes sense because it's completely unbounded)

Feels like we might be able to avoid it for this endpoint given it's only returning currently running ones.  WDYT?

mpawliszyn replied:
https://github.com/cashapp/backfila/blob/master/service/src/main/kotlin/app/cash/backfila/dashboard/GetBackfillRunsAction.kt#L59

mpawliszyn replied:
You are listing runs so of course it needs to be paginated. 

The other ones might need a rethink soon given how popular Backfila has become.

Why did you pick runs to show rather than services?

jrobotham-square commented:
> Mind reposting the screenshots, they don't seem to be showing in the preview?

👍  done.  (not sure what happened there)

jrobotham-square commented:
> Make sure to rebase on top of the variant PR.

I've rebased.  It's not immediately obvious to me how the variant changes work, I've just hardcoded some values to make the tests I added pass.
